core/config_loader.js

```diff
@@ -0,0 +1,240 @@
+//  deps
+const paths = require('path');
+const async = require('async');
+const moment = require('moment');
+
+const _ = require('lodash');
+const mapValuesDeep = require('deepdash/getMapValuesDeep')(_);
+
+module.exports = class ConfigLoader {
```

Comment:
`ConfigLoader` is a generic loader for HJSON configuration files that use includes, @references, etc.